content/learn/book/migration-guides/0.8-0.9/_index.md

```diff
@@ -0,0 +1,830 @@
+
+If using `Option<T>` with Bevy’s reflection API, `T` now needs to implement `FromReflect` rather than just `Clone`. This can be achieved easily by simply deriving `FromReflect`:
+
+```rust
+
+// OLD
+#[derive(Reflect, Clone)]
+struct Foo;
+
+let reflected: Box<dyn Reflect> = Box::new(Some(Foo));
+
+// NEW
+#[derive(Reflect, FromReflect)]
+struct Foo;
+
+let reflected: Box<dyn Reflect> = Box::new(Some(Foo));
+```
+
+Note: You can still derive `Clone`, but it’s not required in order to compile.
+
+### [Remove `ReflectMut` in favor of `Mut<dyn Reflect>`](https://github.com/bevyengine/bevy/pull/5630)
+
+* relax `T: ?Sized` bound in `Mut<T>`
+* replace all instances of `ReflectMut` with `Mut<dyn Reflect>`
+
+### [Update enum derives](https://github.com/bevyengine/bevy/pull/5473)
+
+Bevy-defined enums have been updated to implement `Enum` and are not considered value types (`ReflectRef::Value`) anymore. This means that their serialized representations will need to be updated. For example, given the Bevy enum:
+
+```rust
+pub enum ScalingMode {
+  None,
+  WindowSize,
+  Auto { min_width: f32, min_height: f32 },
+  FixedVertical(f32),
+  FixedHorizontal(f32),
+}
+```
+
+You will need to update the serialized versions accordingly.
+
+```js
+// OLD FORMAT
+{
+  "type": "bevy_render::camera::projection::ScalingMode",
+  "value": FixedHorizontal(720),
+},
+
+// NEW FORMAT
+{
+  "type": "bevy_render::camera::projection::ScalingMode",
+  "enum": {
+    "variant": "FixedHorizontal",
+    "tuple": [
+      {
+        "type": "f32",
+        "value": 720,
+      },
+    ],
+  },
+},
+```
+
+### [remove blanket `Serialize + Deserialize` requirement for `Reflect` on generic types](https://github.com/bevyengine/bevy/pull/5197)
+
+`.register_type` for generic types like `Option<T>`, `Vec<T>`, `HashMap<K, V>` will no longer insert `ReflectSerialize` and `ReflectDeserialize` type data. Instead you need to register it separately for concrete generic types like so:
+
+```rust
+  .register_type::<Option<String>>()
+  .register_type_data::<Option<String>, ReflectSerialize>()
+  .register_type_data::<Option<String>, ReflectDeserialize>()
+```
+
+### [Utility methods for Val](https://github.com/bevyengine/bevy/pull/6134)
+
+Instead of using the + and - operators, perform calculations on `Val`s using the new `try_add` and `try_sub` methods. Multiplication and division remained unchanged. Also, when adding or subtracting from `Size`, ~~use a `Val` tuple instead of `Vec2`~~ perform the addition on `width` and `height` separately.
+
+### [Expose `Image` conversion functions (fixes #5452)](https://github.com/bevyengine/bevy/pull/5527)
+
+* Rename `image_to_texture` to `Image::from_dynamic`
+* Rename `texture_to_image` to `Image::try_into_dynamic`
+* `Image::try_into_dynamic` now returns a `Result` (this is to make it easier for users who didn't read that only a few conversions are supported to figure it out.)

```

Comment:
```suggestion
```

As far as I can tell, no public APIs were harmed in this PR. Looks like you agree in a comment on the PR.